### PR TITLE
Upgrade OpenLayers

### LINKED ISSUE #15296: Upgrade OpenLayers
Currently phpMyAdmin is using [OpenLayers](https://openlayers.org/) version 2.13.1, that was released in 2013.

We should upgrade OpenLayers to the latest version (which at the time of writing is version 5.3.0).

Related to #13173

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

Fixes: #15296

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Review by MauricioFauth — APPROVED** [review]
Looks good to me.

**Comment by williamdes** [comment]
Hi @shucon 
https://github.com/phpmyadmin/phpmyadmin/blob/master/scripts/sync-js-vendor-files.sh must be updated to use node_modules

File: libraries/classes/Gis/GisLineString.php [hunk 1]
```diff
@@ -283,9 +293,10 @@ public function prepareRowAsOl($spatial, $srid, $label, $line_color, array $scal
             );
         $points_arr = $this->extractPoints($linesrting, null);
 
-        return $result . 'vectorLayer.addFeatures(new OpenLayers.Feature.Vector('
-            . $this->getLineForOpenLayers($points_arr, $srid)
-            . ', null, ' . json_encode($style_options) . '));';
+        return $result . 'var line = new ol.Feature({geometry: '
+            . $this->getLineForOpenLayers($points_arr, $srid) . '});'
+            . 'line.setStyle(style);'
+            . 'vectorLayer.addFeature(line);';
```
Comment (williamdes) [inline libraries/classes/Gis/GisLineString.php:299]: Please use back `json_encode` :)

File: libraries/classes/Gis/GisMultiLineString.php [hunk 2]
```diff
@@ -313,10 +323,11 @@ public function prepareRowAsOl($spatial, $srid, $label, $line_color, array $scal
...
 
-        return $row . 'vectorLayer.addFeatures(new OpenLayers.Feature.Vector('
-            . 'new OpenLayers.Geometry.MultiLineString('
-            . $this->getLineArrayForOpenLayers($linestirngs, $srid)
-            . '), null, ' . json_encode($style_options) . '));';
+        return $row . $this->getLineArrayForOpenLayers($linestirngs, $srid)
+            . 'var multiLineString = new ol.geom.MultiLineString(arr);'
+            . 'var feature = new ol.Feature({geometry: multiLineString});'
+            . 'feature.setStyle(style);'
+            . 'vectorLayer.addFeature(feature);';
```
Comment (williamdes) [inline libraries/classes/Gis/GisMultiLineString.php:330]: same here

**Comment by shucon** [comment]
If we want to run the code using yarn, we might have to change the code to something like this https://www.npmjs.com/package/ol to import each library.

**Comment by williamdes** [comment]
> If we want to run the code using yarn, we might have to change the code to something like this https://www.npmjs.com/package/ol to import each library.

They do not seem to distribute bundles files..
Just add a comment in the sync file then :)

File: libraries/classes/Gis/GisLineString.php (hunk 1 above)
Comment (shucon) [reply libraries/classes/Gis/GisLineString.php:299]: `json_encode` type input is not supported in the current OL format. If you'll look at how styling is coded now, you'll understand.

File: templates/gis_data_editor_form.twig [hunk 3]
```diff
@@ -219,3 +219,4 @@
         </div>
     </div>
 </form>
+<link rel="stylesheet" href="js/vendor/openlayers/theme/ol.css">
```
Comment (williamdes) [inline templates/gis_data_editor_form.twig:222]: This seems like a hack 
What do you think @mauriciofauth ?

Comment (mauriciofauth) [reply templates/gis_data_editor_form.twig:222]: Probably better to add it in another place.

**Comment by shucon** [comment]
Does the json_encode looking styling seems good now? @williamdes

**Review by MauricioFauth — APPROVED** [review]

**Comment by shucon** [comment]
Can this be merged now? Else there would be conflicts popping up.

File: libraries/classes/Gis/GisGeometry.php [hunk 4]
```diff
@@ -336,25 +333,21 @@ protected function getLineArrayForOpenLayers(
...
-            $ol_array .= $this->getLineForOpenLayers(
+            $ol_array .= 'var line = ' . $this->getLineForOpenLayers(
                 $points_arr,
                 $srid,
                 $is_line_string
-            );
-            $ol_array .= ', ';
+            ) . ';';
+            $ol_array .= 'var coord = line.getCoordinates();';
+            $ol_array .= 'coord.forEach(item => lineArr.push(item));';
```
Comment (williamdes) [inline libraries/classes/Gis/GisGeometry.php:346]:
```suggestion
            $ol_array .= 'coord.forEach(item => lineArr.push(item));';
```

Oops you can not use such features..
https://caniuse.com/#search=forEach
https://caniuse.com/#feat=arrow-functions

File: libraries/classes/Gis/GisVisualization.php [hunk 5]
```diff
@@ -459,39 +482,31 @@ public function asOl()
...
-        $output .= 'map.addLayer(vectorLayer);'
-            . 'map.zoomToExtent(bound);'
-            . 'if (map.getZoom() < 2) {'
-            . 'map.zoomTo(2);'
-            . '}'
-            . 'map.addControl(new OpenLayers.Control.LayerSwitcher());'
-            . 'map.addControl(new OpenLayers.Control.MousePosition());'
+        $output = 'if (typeof ol !== "undefined") {'
+            . '$(\'head\').append(\'<link rel="stylesheet" type="text/css"'
+            . ' href="js/vendor/openlayers/theme/ol.css">\');'
```
Comment (williamdes) [inline libraries/classes/Gis/GisVisualization.php:487]: I would suggest to use a constant to store the css path in a more dynamic way

**Review by williamdes — APPROVED** [review]
LGTM

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**